arlas-core/src/main/java/io/arlas/server/managers/CollectionReferenceManager.java

```diff
@@ -55,15 +56,29 @@ public ElasticType getType(CollectionReference collectionReference, String field
         if (elasticType == null) {
             String[] props = field.split("\\.");
             CollectionReferenceDescriptionProperty esField = elasticAdmin.describeCollection(collectionReference).properties.get(props[0]);
+            if (esField == null) {
+                if (throwException) {
+                    throw new NotFoundException("Field '" + field + "' not found in collection " + collectionReference.collectionName);
+                } else {
+                    return ElasticType.UNKNOWN;
+                }
+            }
             for (int i=1; i<props.length; i++) {
                 esField = esField.properties.get(props[i]);
+                if (esField == null) {
```

Comment:
It looks like you doing the same check 2 times, can that be put in a dedicated method?

Reply:
absolutely

Reply:
fixed